### ISSUE TITLE
Issues sending and receiving payments

### ISSUE THREAD

**Issue by mat7ias** [raiden-network/raiden#3233]
- [x] The Ethereum node was always up to date with the chain tip and never stopped running.
- [x] No Ethereum transaction was sent for the account that Raiden manages by another app.
- [x] The Ethereum account used by Raiden always had sufficient ETH.
- [x] The Raiden DB that exists by default at ~/.raiden was never altered or deleted.
- [ ] The Raiden node was never restarted. If yes please provide more info.
- [x] The ethereum node was not switched while Raiden was running.
- [x] The Raiden REST API is protected from public access.

-->

## Problem Definition

I was trying to interact with cryptoplayer.one and also @loredanacirstea on the mainnet WETH token network but had issues getting payments to go through (sending and receiving). I tried multiple payments to 0xFeA60432EEb8E858364E0f8953F1386e853A7804 (cryptowars game guardian) which weren't successful. It didn't find a suitable path. I tried restarting my node to see if that'd fix the issue. I also had payments that didn't go through to my node.

[raiden-debug_2019-01-05T10#003A05#003A26.451552.log](https://github.com/raiden-network/raiden/files/2729463/raiden-debug_2019-01-05T10.003A05.003A26.451552.log)

I shared the db file with loredana and it's forwarded to @LefterisJP 

### System Description

Raiden Red Eyes (v100.0.1), Parity Ethereum 2.1.10-stable

**Comment by loredanacirstea** [comment]
I (`0x6a7b31479Bff6cE4d01ea6BCf2E7ca458e0B40d1`) opened a channel + deposit with @mat7ias (`0x362708Dd5e81A85e75E877dF718A0A9eF35e8267`) and then tried to send two payments from my side. Both expired.

[raiden-debug_2019-01-05T09:23:29.135362.log](https://github.com/raiden-network/raiden/files/2729480/raiden-debug_2019-01-05T09.23.29.135362.log)

(using the same Raiden version v100.0.1)

**Comment by LefterisJP** [comment]
@andrevmatos and @ulope can you take a look at the log and tell me why would Matthias always be seeing `Partner not reachable. Skipping.` whenever he tries to send a payment to anyone? We know that the partner is online.

**Comment by ulope** [comment]
Ok, looked at the logs.

In @mat7ias' log not a singe `Changing address presence state` or `Incoming messages` event was logged. That combined with the fact that @loredanacirstea's log showed @mat7ias as offline suggests that something was wrong with the matrix server @mat7ias was connected to (which was exchange union in this case).

I'll try to see if I can find something in the monitoring. 
Also we should do some more tests with one user on that specific server to rule out that there is a persistent issue.

**Comment by loredanacirstea** [comment]
I suggested @mat7ias to change the matrix server and: 

> sending payments seems to be successful now that I'm using a different transport server

**Comment by LefterisJP** [comment]
A user in gitter also got the same with the exchange union matrix server: https://gitter.im/raiden-network/raiden?at=5c3be3899fa58135b55fb063

How can we determine if it is indeed a problem of that server?

**Comment by kilrau** [comment]
Might be a problem when two nodes are connected to two *different* matrix servers, that's what we experienced in our tests @mat7ias . When both nodes use `raidentransport.exchangeunion.com`, we don't see these problems.

Could you re-try your setup with both nodes being connected to `raidentransport.exchangeunion.com`? Because if there is indeed a problem with our server, we'd want to fix that.

**Comment by mat7ias** [comment]
@kilrau I followed the setup you suggested and wasn't able to create the issue when I had both nodes use ```raidentransport.exchangeunion.com``` so I think you might be correct. 
However, I'm using the v0.100.3 (Rosemary) release now, different from the v0.100.1 when I first encountered the issue.

I'm able to recreate what I'd assume is the same issue with v0.100.3.
I switched between having both nodes use ```raidentransport.exchangeunion.com``` (transfers fine)
Then ```raidentransport.exchangeunion.com``` and ```raidentransport.mycryptoapi.com``` (issue is created)
